### DISCUSSION TITLE
Discharging on turn on, when connected to XG Mobile

### DISCUSSION THREAD

**Discussion by Chrono-dad** [seerge/g-helper#1299] [Q&A]
I'm having an issue, having my Rog Flow X13 connected to the XG Mobile. Its charged/charging the night before then turn off, in morning turn on and it starts charged how I have it set at 80% and then discharges. Only way round this is to cancel connection to XG Mobile. Unplug, plug back in and restart connection. Sometimes this doesn't work and I need to shut down laptop and disconnect cable. Also have noticed that red light on XG mobile is not  staying on either at anytime.

**Comment by seerge** [comment]
@Chrono-dad hello, I don't have X13 / XGM, but I can only assume that in terms of charging XGM acts as a regular usb-c charger. 

G-Helper doesn't control charging anyhow - it's taken care of on hardware level (the only thing that can be configured here is charge limit)

**Comment by Chrono-dad** [comment]
So I stopped g-helper and reinstalled Armory Crate to see what the difference is, so armoury crate is what controls the light on the front of the XGM. Also the problem I had with g-helper where if I turned off the laptop with XGM connected and turned back on the battery just discharges, with armoury crate its not doing that at all. I turned off all day and turned on, found that my battery fully charged and staying charged (showing plugged in)
________________________________

**Comment by seerge** [comment]
@Chrono-dad may be there are other unknown commands that tell XGM to start charging then . 

You can try to run activate / deactivate command for XGM manually from powershell as admin (but w/o AC for a clean test)

To Activate XGM (it's known and should work)
```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090019, 1)
```

To Deactivate XGM (it's known and should work)
```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090019, 0)
```

And then after deactivating try to activate it with each of the following and see if it affects power limit

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090019, 0x0101)
```
```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090019, 0x0201)
```
```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090019, 0x0301)
```

And see if activating with 0x0101 or 0x0201 makes difference

**Reply by seerge** [reply]
@Chrono-dad  P.S. For XGM light - there is a separate switch for that under Extra settings in G-Helper, does checking / unchecking it change light status ?

**Comment by Chrono-dad** [comment]
Sorry late getting back to you, yes is the answer to the XGM back light. the selection was on, when unchecked and checked the light came on.

I've also noticed your g-helper 0.23.0 update; again, sorry I've been preoccupied and didn't have a moment to try your suggestions.

I hope your update works though.

Thankyou Serge for your help
________________________________
